### ISSUE TITLE
Can't create video player on Ubuntu 15.10 and 16.04

### ISSUE THREAD

**Issue by nicolaspayette** [NetLogo/vid#9]
In a blank model:
```netlogo
extensions [ vid ]

to setup
  clear-all
  vid:movie-select
end
```

When choosing any `.mp4` file, it gives:

```
MediaException: UNKNOWN : com.sun.media.jfxmedia.MediaException: Could not create player! : com.sun.media.jfxmedia.MediaException: Could not create player!
 at javafx.scene.media.MediaException.exceptionToMediaException(MediaException.java:146)
 at javafx.scene.media.MediaPlayer.init(MediaPlayer.java:511)
 at javafx.scene.media.MediaPlayer.<init>(MediaPlayer.java:414)
 at org.nlogo.extensions.vid.Movie$.buildMovie(Movie.scala:67)
 at org.nlogo.extensions.vid.Movie$.open(Movie.scala:39)
 at org.nlogo.extensions.vid.MovieSelect$$anonfun$perform$1.apply(MovieSelect.scala:12)
 at org.nlogo.extensions.vid.MovieSelect$$anonfun$perform$1.apply(MovieSelect.scala:10)
 at scala.Option.foreach(Option.scala:257)
 at org.nlogo.extensions.vid.MovieSelect.perform(MovieSelect.scala:10)
 at org.nlogo.prim._extern.perform(_extern.java:38)
 at org.nlogo.nvm.Context.stepConcurrent(Context.java:91)
 at org.nlogo.nvm.ConcurrentJob.step(ConcurrentJob.java:83)
 at org.nlogo.job.JobThread.org$nlogo$job$JobThread$$runPrimaryJobs(JobThread.scala:133)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply$mcV$sp(JobThread.scala:68)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply(JobThread.scala:66)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply(JobThread.scala:66)
 at scala.util.control.Exception$Catch.apply(Exception.scala:103)
 at org.nlogo.api.Exceptions$.handling(Exceptions.scala:41)
 at org.nlogo.job.JobThread.run(JobThread.scala:65)
Caused by: com.sun.media.jfxmedia.MediaException: Could not create player!
 at com.sun.media.jfxmediaimpl.NativeMediaManager.getPlayer(NativeMediaManager.java:274)
 at com.sun.media.jfxmedia.MediaManager.getPlayer(MediaManager.java:118)
 at javafx.scene.media.MediaPlayer.init(MediaPlayer.java:467)
 ... 17 more

NetLogo 6.0-M8
main: org.nlogo.app.AppFrame
thread: JobThread
Java HotSpot(TM) 64-Bit Server VM 1.8.0_91 (Oracle Corporation; 1.8.0_91-b14)
operating system: Linux 4.4.0-31-generic (amd64 processor)
Scala version 2.11.8
JOGL: (3D View not initialized)
OpenGL Graphics: (3D View not initialized)
model: Untitled

03:24:46.947 SwitchedTabsEvent (org.nlogo.app.Tabs) AWT-EventQueue-0
03:24:46.946 RuntimeErrorEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
03:24:46.946 PeriodicUpdateEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
03:24:35.722 PeriodicUpdateEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
03:24:35.721 TickStateChangeEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) JobThread
03:24:35.721 OutputEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
03:24:35.719 AddJobEvent (org.nlogo.app.CommandLine) AWT-EventQueue-0
03:24:35.718 OutputEvent (org.nlogo.app.CommandLine) AWT-EventQueue-0
03:24:35.718 CompiledEvent (org.nlogo.window.CompilerManager) AWT-EventQueue-0
03:24:35.708 CompileMoreSourceEvent (org.nlogo.app.CommandLine) AWT-EventQueue-0
```

It _could_ be something to do with my configuration, but I tried on both my 15.10 laptop and a fresh install of 16.04 on a desktop, with the `libavcodec-extra` package installed.

Possibly relevant SO questions include:

http://stackoverflow.com/questions/24090356/javafx-mediaplayer-could-not-create-player-error-in-ubuntu-14-04

and

http://stackoverflow.com/questions/38164044/javafx-media-player-on-ubuntu-16-04

where the latter (worryingly) states that "ubuntu 16.04 was not among supported operating systems for javafx 1.8", but I didn't find a primary source to confirm that information.

Has anyone been able to open a movie using `vid` in Linux?

**Comment by mrerrormessage** [comment]
I haven't tested it personally on linux. The (badly outdated) [system requirements page](http://docs.oracle.com/javafx/2/system_requirements_2-2-5/jfxpub-system_requirements_2-2-5.htm) suggests that it requires `libavcodec` and `libavformat` (and perhaps a specific version, at that). I'm assuming you're working on 64-bit linux with 64-bit Java/NetLogo?

**Comment by nicolaspayette** [comment]
Yup:

    Java HotSpot(TM) 64-Bit Server VM 1.8.0_91 (Oracle Corporation; 1.8.0_91-b14)
    operating system: Linux 4.4.0-31-generic (amd64 processor)

I didn't specify, but that was running from the main `hexy` branch.

**Comment by nicolaspayette** [comment]
`libavformat-ffmpeg56` (7:2.8.6-1ubuntu2) is installed in addition to `libavcodec-extra` (7:2.8.6-1ubuntu2).

**Comment by mrerrormessage** [comment]
For what it's worth, I am able to get it working on my Ubuntu 14.04 VM. [this gist](https://gist.github.com/mrerrormessage/cc776627624c02f6bd904139eb032f7d) has all the packages on the machine that start with `libav` as well as all of the `.so` files that java had open while running the vid player. It might be worth comparing to your list. Note that I happened to use java 1.8u74 because it was on the path, but I wouldn't expect significant changes between 1.8u74 and 1.8u92 on this front.

**Comment by nicolaspayette** [comment]
The only difference between my list of packages and yours is that I had different (more recent) versions of `libavcodec54`, `libavformat54` and `libavutil52`. I tried manually installing the `*54` libs by using instructions [found here](http://askubuntu.com/questions/665662/how-to-install-libavcodec54-in-ubuntu-vivid), but the end result is the same.

My (now updated) list of `libav*` packages is [here](https://gist.github.com/nicolaspayette/7709565e2e262f00a6ca6ba9e0048bb5).

Comparing the content of `/lib/x86_64-linux-gnu` yields too many differences for me to make sense of it. FWIW, my dump is [here](https://gist.github.com/nicolaspayette/e6fd0301a786b32eeea8b9767cd4007c).

Not sure what to try next...

**Comment by mrerrormessage** [comment]
After a very thorough investigation, Nicolas and I determined that the cause was an incompatible version of libavformat. Conveniently, there is [already a java bug for this](https://bugs.openjdk.java.net/browse/JDK-8154378) which has a fix slated for release in java 8u102. 8u102 is due out in July (so hopefully soon). I'm going to leave this open until we can confirm the issue is fixed. If it isn't fixed, we'll want to consider a workaround of some kind.

**Comment by mrerrormessage** [comment]
JDK 8u102 was released today. I'm going to try to put out a new build with that version bundled tomorrow, which should give us a good idea whether this is fixed.

**Comment by nicolaspayette** [comment]
I confirmed that this is fixed when using 6.0-M9.
